crypto/bn/bn_lib.c

```diff
@@ -179,20 +179,17 @@ static void bn_free_d(BIGNUM *a)
 
 void BN_clear_free(BIGNUM *a)
 {
-    int i;
-
     if (a == NULL)
         return;
     bn_check_top(a);
```

Comment:
Formally speaking this bn_check_top should be moved inside the if below, because the kind of thing it does can't be performed on STATIC_DATA. Though I fail to see what purpose does it serve, since we are about to free it anyway. Well, there is assert... Otherwise it can as well be omitted... I mean move or omit :-)

Reply:
Same can be said about BN_free. I.e. move or omit bn_check_top. Just in case assert itself is not the problem, there is bn_pollute. On the other hand, one might be able to argue that STATIC_DATA should be perfectly trimmed, i.e. with top being equal to dmax and with topmost limb being non-zero, so that one might appreciate SEGV as indication that it's not right...

Reply:
Yes I wasn't sure what purpose bn_check_top server either. I'd removed both of them.